code/modules/surgery/surgery.dm

```diff
@@ -129,10 +141,16 @@
 	if(iscyborg(user))
 		var/mob/living/silicon/robot/R = user
 		var/obj/item/surgical_processor/SP = locate() in R.module.modules
-		if(!SP)
-			return FALSE
-		if(type in SP.advanced_surgeries)
-			return TRUE
+		if(!isnull(SP))
+			if(type in SP.advanced_surgeries)
```

Comment:
Someone might say `if(!isnull(SP) && type in SP.advanced_surgeries)`. I don't think that's a good condition because `if(!isull(N))` is important check more than `if(type in surgeries)`. I'll change that if you want that double condition in a line.